modules/server/src/main/scala/scaladex/server/route/ProjectPages.scala

```diff
@@ -111,6 +111,37 @@ class ProjectPages(env: Env, database: WebDatabase, searchEngine: SearchEngine)(
           }
         }
       },
+
+      get {
+        path(projectM / "artifacts" ) { ref => 
+          getProjectOrRedirect(ref,user) { project => 
+            val headerF = service.getProjectHeader(project)
+            for {
+              header <- headerF
+            } yield {
+              val allArtifacts = header.toSeq.flatMap(_.artifacts)
+
+              // Group by artifact name
+              val artifactsByName = allArtifacts.groupBy(_.artifactName)
+
+              val groupedArtifacts = artifactsByName
+                .groupBy { case (artifactName, artifacts) =>  
+                  artifacts.maxBy(_.version).version 
+                }
+                .map { case (latestVersion, artifactsByName) =>
+                  latestVersion -> artifactsByName.map { case (name, artifacts) => name -> artifacts.filter(_.version == latestVersion)
+                }
+              }
```

Comment:
Then you can use the params to filter the artifact groups. You want to keep the groups which contains at least one artifact for each required binary version.

```suggestion
              val groupedArtifacts = artifactsByName
                .groupBy { case (artifactName, artifacts) =>  
                  artifacts.maxBy(_.version).version 
                }
                .map { case (latestVersion, artifactsByName) =>
                  latestVersion -> artifactsByName
                    .map { case (name, artifacts) => name -> artifacts.filter(_.version == latestVersion) }
                    .filter { case (_, artifacts) =>
                      params.binaryVersions.forall(binaryVersion => artifacts.exists(_.binaryVersion == binaryVersion))
                    }
            }
```